File: 2019/php/src/Aoc/Days/Day2.php
<?php

namespace Ppx17\Aoc2019\Aoc\Days;

use Ppx17\Aoc2019\Aoc\Days\Day2\IntCode;

class Day2 extends AbstractDay
{
    private IntCode $intCode;

    public function setUp(): void
    {
        parent::setUp();

        $this->intCode = new IntCode(array_map('intval', explode(',', $this->getInput())));
    }

    public function dayNumber(): int
    {
        return 2;
    }

    public function part1(): string
    {
        $this->intCode->memory[1] = 12;
        $this->intCode->memory[2] = 2;
        return (string)$this->intCode->run();
    }

    public function part2($target = 19690720): string
    {
        for ($noun = 99; $noun >= 0; $noun--) {
            for ($verb = 99; $verb >= 0; $verb--) {
                $this->intCode->reset();
                $this->intCode->memory[1] = $noun;
                $this->intCode->memory[2] = $verb;
                $result = $this->intCode->run();
                if ($result === $target) {
                    return (string)(100 * $noun + $verb);
                }
            }
        }
        return '';
    }
}

File: 2019/julia/src/Day2.jl
module Day2

include("Aoc.jl")
using .Aoc
include("Intcode.jl")
using .Intcode

intcode = Aoc.intcode(2)

function run(noun, verb)
	code = copy(intcode)
	code[2] = noun
	code[3] = verb
	comp = Intcode.Computer(code)
	Intcode.run!(comp)
end

part1() = run(12, 2)

function part2()
	for noun in 1:100, verb in 1:100
		run(noun, verb) == 19690720 && return 100noun + verb
	end
end

export part1, part2
end